test-data/unit/fixtures/dict.pyi

```diff
@@ -1,5 +1,6 @@
 # Builtins stub used in dictionary-related test cases.
 
+from _collections_abc import dict_keys
```

Comment:
This line looks to be the cause of I think all of the test failures, e.g.

```
$ pytest -q mypy/test/testcheck.py::TypeCheckSuite::check-incremental.test::testIncrementalWithDifferentKindsOfNestedTypesWithinMethod 
F                                                                                                                                                       [100%]
========================================================================== FAILURES ===========================================================================
_________________________________________________ testIncrementalWithDifferentKindsOfNestedTypesWithinMethod __________________________________________________
data: /home/mjh/src/mypy/test-data/unit/check-incremental.test:5662:
/home/mjh/src/mypy/mypy/test/testcheck.py:80: in run_case
    self.run_case_once(testcase, ops, step)
/home/mjh/src/mypy/mypy/test/testcheck.py:174: in run_case_once
    assert_string_arrays_equal(output, a, msg.format(testcase.file, testcase.line))
E   AssertionError: Unexpected type checker output in incremental, run 2 (/home/mjh/src/mypy/test-data/unit/check-incremental.test, line 5662)
-------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------
Expected:
  tmp/a.py:2: error: "object" has no attribute "xyz" (diff)
Actual:
  tmp/a.py:2: error: Module has no attribute "xyz" (diff)

Alignment of first line difference:
  E: tmp/a.py:2: error: "object" has no attribute "xyz"
  A: tmp/a.py:2: error: Module has no attribute "xyz"
                        ^
=================================================================== short test summary info ===================================================================
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-incremental.test::testIncrementalWithDifferentKindsOfNestedTypesWithinMethod
1 failed in 0.33s
```

I'm going to have a look around, but would be happy to hear any suggestions :upside_down_face:

Reply:
Presumably the stubs used for these tests don't include stubs for `_collections_abc`. As I recall there are stubs for various important standard library modules somewhere in the fixtures directory.

This may be difficult to get right as we have had some instability around the type of `dict.keys()` before; what you're using here (`_collections_abc.dict_keys`) is arguably a private implementation detal.

Reply:
> This may be difficult to get right as we have had some instability around the type of `dict.keys()` before; what you're using here (`_collections_abc.dict_keys`) is arguably a private implementation detail.

:+1: I'm also not 100% on the approach, so happy to hear any alternatives you might have. So far I've just been following typeshed https://github.com/python/typeshed/blob/master/stdlib/builtins.pyi#L1039